cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -422,6 +426,93 @@ impl YaccParser {
         }
     }
 
+    // Handle parse_param declarations of the form:
+    // %parse_param <'a>(x: u32, y : (u32, u32))
+    fn parse_param(&mut self, mut i: usize) -> YaccResult<usize> {
+        i = self.parse_ws(i, false)?;
+        // First gobble up all of the '<' lifetime ',' ... '>
+        if let Some(mut j) = self.lookahead_is("<", i) {
+            let mut k = j;
+            let mut lifetimes = HashSet::new();
+            let mut add_lifetime = |k, c: char| {
+                let s = self.src[j..k].trim_start().trim_end().to_string();
+                lifetimes.insert(s);
+                j = k + c.len_utf8();
```

Comment:
Can replace this line and the one below with just `k + c.len_utf8()`.

Reply:
This was actually updating j, in the loop from where `add_lifetime` is called, 
I think its clearer with the above change, and mutation moved out.

Sorry it has been some years since i've done much imperative programming, and it tells.